sys/syz-extract/linux.go

```diff
@@ -89,11 +89,17 @@ func (*linux) prepareArch(arch *Arch) error {
 	buildDir := arch.buildDir
 	makeArgs := []string{
 		"ARCH=" + target.KernelArch,
-		"CROSS_COMPILE=" + target.CCompilerPrefix,
 		"CFLAGS=" + strings.Join(cflags, " "),
 		"O=" + buildDir,
 		"-j", fmt.Sprint(runtime.NumCPU()),
 	}
+	if target.Triple != "" {
```

Comment:
It would bee good to move as many if's and distinction between gcc/clang from this file to sys/targets. All of this logic will be needed in pkg/build eventually as it also builds kernels. And this tool should preferably not concerned with differences between compilers as much as possible in general.

Triple seems to be the right thing to specify to specify in config, but it seems that nobody outside of sys/targets will actually use it (and should not! for clang it's already in compiler flags). What users need seems to be still CCompilerPrefix. So I think prefix should be unexported, affect cflags and be exported in CCompilerPrefix form.

This:
```
	if arch.useClang {
		makeArgs = append(makeArgs, "CC=clang")
	}
```
should be (in the argument list above):
```
    "CC=" + target.Something,
```


And this chunk below:
```
	var cc string
	if !arch.useClang {
		cc = "gcc"
		if arch.target.Triple != "" {
			cc = arch.target.Triple + "-" + cc
		}
	} else {
		cc = "clang"
	}
```
should be:
```
cc := arch.target.Something
```

Reply:
CCompilerPrefix is a gcc-ism, Clang doesn't require a prefix. Instead, as you suggested above, let's export e.g. TargetCompiler.

Reply:
Actually, we already have target.CCompiler that seems to do what we need.